app/src/main/java/unipd/se18/ocrcamera/BlurGalleryActivity.java

```diff
@@ -0,0 +1,143 @@
+package unipd.se18.ocrcamera;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+
+import static android.graphics.BitmapFactory.decodeStream;
+
+/** Testing Activity to see if blur could work, shows a gallery with each image associated with it's blur value
+ * Leonardo Pratesi - gruppo 1
+ * Activity accessible by the options menu on the application
+ *
+ */
+public class BlurGalleryActivity extends AppCompatActivity {
+
+    /**
+     *  Directory of the Photos
+     */
+    private final String PHOTOS_FOLDER = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/OCRCameraDB";
+
+
+    //private final String PHOTOS_FIX_ERROR = "/storage/emulated/0/Images/OCRCameraDB";
+    /**
+     * ListView to show all the ingredients in a listllayout
+     */
+    private ListView listView;
+
+    /**
+     * ArrayList of object (see class blur object)
+     */
+    private ArrayList<BlurObject> arrayBlur = new ArrayList<>();
+
+    /**
+     * listllayout of elements inside PHOTOS_FOLDER
+     */
+    private String fileNames[];
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setContentView(R.layout.listllayout);
+        super.onCreate(savedInstanceState);
+
+        File path = new File(PHOTOS_FOLDER);
+        fileNames = path.list();
+        if (fileNames != null) {
+            setGallery();
+        }
+        else {
+            Toast.makeText(this, "No images found", Toast.LENGTH_SHORT).show();
+            this.finish();
+            //Exit from Gallery
+        }
+
+        //View preparation
+        listView = (ListView) findViewById(R.id.listview);
+        BlurObjectAdapter adapter = new BlurObjectAdapter(this, R.layout.listllayout, arrayBlur);
+        listView.setAdapter(adapter);
+        //show max blur value
+        TextView maxview = findViewById(R.id.textViewMax);
+        maxview.setText(String.valueOf(findMax(arrayBlur)));
+
+
+    }
+
+    /**
+     * Method that creates a blurObject for every image in the folder
+     *
+     */
+    private void setGallery() {
+        int count =0;
+        for (int i = 0; i < fileNames.length; i++) {                                                          //imagepath.length = number of elements in the folder

```

Comment:
When iterating on a list you can use a "foreach" which makes the code more readable. The syntax in java would be: foreach(String fileName : fileNames) {...}.

Anyway you can directly iterate on each file using File.listFiles() (applied on Photo folder) that return a File[] array.